`vis.gl` Charter and Governance Proposal for OpenJS (#13)
visgl/tsc: CHARTER.md

# vis.gl Charter

## Section 0: Guiding Principles

The `vis.gl` project is part of the [OpenJS Foundation][https://openjsf.org], which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.

All collaborators must allow open participation from any individual or organization meeting the requirements for contributing under this Charter and any policies adopted by the TSC, regardless of competitive interests. The community must not exclude any participant based on criteria other than those applied consistently and reasonably to all collaborators.

## Section 1: Scope

`vis.gl` is to provide a comprehensive, open-source solution for GPU-powered visualizations of large-scale datasets, in particular geospatial data visualizations.

> The scope of project includes documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source project.

Author: Can we cut this paragraph? It's redundant to the section right below it.

 - Formation of technical **working groups**.
 - **Representation** in technical collaborations with external communities.

### 1.2: Out-of-Scope

Section Intentionally Left Blank

## Section 2: Relationship with OpenJS Foundation CPC.

Technical leadership for `vis.gl` is delegated by the OpenJS Cross-Project Council (CPC) through this charter. The `vis.gl` Technical Steering Committee (TSC) is responsible for all technical aspects of the vis.gl project and its sub-projects. The OpenJS Foundation's business leadership is the Board of Directors (the “Board”).

The TSC operates independently on technical matters. This charter may be amended by the TSC (via a majority vote) subject to CPC approval. The CPC may also propose amendments, but does not interfere with day-to-day technical decisions.

### 2.1 Other Formal Project Relationships

Section Intentionally Left Blank

## Section 3: vis.gl's TSC Governing Body

vis.gl is governed by its Technical Steering Committee (TSC). The current list of TSC members is maintained in the vis.gl organization’s GitHub repository.

## Section 4: Roles & Responsibilities 

> The TSC oversees the technical scope outlined in Section 1 and may delegate tasks to Maintainers and Contributors. Details on membership, roles, responsibilities, and decision-making are defined in the `GOVERNANCE.md` document.

Author: ```suggestion
The Steering Committee Members oversees the technical scope outlined in Section 1 and may delegate tasks to Maintainers and Contributors. Details on membership, roles, responsibilities, and decision-making are defined in the `GOVERNANCE.md` document.
```

# vis.gl Charter

> 

Author: Pulling from our [previous iteration](https://docs.google.com/document/d/1djDQfY3TMoUuh22PdJBUm6gMHu-PW4Hf25rpHfg962o/edit?tab=t.0).. something like

```suggestion
Vis.gl aims to provide an open-source JavaScript framework for high-performance visualizations of large datasets on the web. It prioritizes strong support for geospatial data visualization, while also being flexible to cater to diverse non-geospatial visualization needs. It is designed to make advanced visualization techniques accessible to a wide range of developers by providing a user-friendly API and well-documented framework.
```

## Section 0: Guiding Principles

> The `vis.gl` project is part of the [OpenJS Foundation][https://openjsf.org], which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.

Reviewer: ```suggestion
The `vis.gl` project is part of the [OpenJS Foundation](https://openjsf.org), which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.
```

The scope of project includes documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source project.

### 1.1: In-scope

 - Overseeing the **technical development** of the core vis.gl libraries and all affiliated sub-projects.
 - **Release planning** and coordination
 - Managing project repositories and **governance**, including issue tracking, pull request processes, and repository organization.
 - Defining and enforcing **coding standards, testing practices, and development processes** to maintain code quality.
 - Maintaining the list of **Maintainers and core contributors**, and setting permissions in project repositories.
 - Reviewing and proposing **new project incubations or sub-projects**, as well as deciding on deprecations or significant changes to the project’s scope.
 - Formation of technical **working groups**.
 - **Representation** in technical collaborations with external communities.

### 1.2: Out-of-Scope

Section Intentionally Left Blank

## Section 2: Relationship with OpenJS Foundation CPC.

> Technical leadership for the projects within the [OpenJS Foundation](https://openjsf.org/) is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the `vis.gl` project, it is delegated to the [Steering Committee Members](https://github.com/visgl/tsc/blob/master/README.md#steering-committee-members) (the “TSC”). The OpenJS Foundation's business leadership is the Board of Directors (the “Board”).

Reviewer: ```suggestion
Technical leadership for the projects within the [OpenJS Foundation](https://openjsf.org/) is delegated to the projects through their project charters by the [OpenJS Foundation Cross-Project Council](https://openjsf.org/about/governance/) (CPC). In the case of the `vis.gl` project, it is delegated to the [Technical Steering Committee](https://github.com/visgl/tsc/blob/master/README.md#steering-committee-members) (the “TSC”). The OpenJS Foundation's business leadership is the Board of Directors (the “Board”).
```

This `vis.gl` Charter reflects a carefully constructed balanced role for the TSC and the CPC in the governance of the OpenJS Foundation. The charter amendment process is for the TSC to propose changes using simple majority of the full TSC, the proposed changes being subject to review and approval by the CPC. The CPC may additionally make amendments to the project charter at any time, though the CPC will not interfere with day-to-day discussions, votes, or meetings of the TSC.

### 2.1 Other Formal Project Relationships

Section Intentionally Left Blank

## Section 3: `vis.gl`'s TSC Governing Body

> `vis.gl` is governed by its [Steering Committee Members](https://github.com/visgl/tsc/blob/master/README.md#steering-committee-members).

Reviewer: Either use TSC or SCM, but don't do both.

Also, it is the governing body that has decision making power, not its members.

Author: Ok, this is something I was unsure of. Other charters like NVM use "Maintainers", the role, instead of a TSC, a body.

I'm on board with using TSC in the charter. What is the way a TSC usually differentiates the people who are providing leadership and voting responsibility vs the people who are only delegated some responsibilities? We current define the leadership group "Steering committee members" and the other group "Maintainers" - is this fine or is it unclear as-written?

Reviewer: I'm just suggesting to go for consistency here. The charter uses "Steering Committee", "TSC", and "Steering Committee Members." Just pick one and go with it. I'd advise using TSC as that's what others do.

## Section 4: Roles & Responsibilities 

The roles and responsibilities of `vis.gl`'s TSC are described in [GOVERNANCE.md](./GOVERNANCE.md).

### Section 4.1 Project Operations & Management

Section Intentionally Left Blank

### Section 4.2: Decision-making, Voting, and/or Elections

> The project will define a consensus-seeking governance model in [GOVERNANCE.md](./GOVERNANCE.md) as need arises.

Reviewer: I'd rather this used the CPC's one instead of punting to later.

Author: I don't think we're punting this - we have a decision making process defined in the governance.

Could you clarify what you're recommending? I had pulled this from Jest's charter (also used in NVM and some others). I'm curious when we'd want to not align with the other projects?

Author: It's added to the charter

# vis.gl Charter
Vis.gl aims to provide an open-source JavaScript framework for high-performance visualizations of large datasets on the web. It prioritizes strong support for geospatial data visualization, while also being flexible to cater to diverse non-geospatial visualization needs. It is designed to make advanced visualization techniques accessible to a wide range of developers by providing a user-friendly API and well-documented framework.
## Section 0: Guiding Principles

The `vis.gl` project is part of the [OpenJS Foundation](https://openjsf.org), which operates transparently, openly, collaboratively, and ethically. Project proposals, timelines, and status must not merely be open, but also easily visible to outsiders.

> All collaborators must allow open participation from any individual or organization meeting the requirements for contributing under this Charter and any policies adopted by the TSC, regardless of competitive interests. The community must not exclude any participant based on criteria other than those applied consistently and reasonably to all collaborators.

Author: ```suggestion
```

## Section 1: Scope

`vis.gl` is to provide a comprehensive, open-source solution for GPU-powered visualizations of large-scale datasets, in particular geospatial data visualizations. The `vis.gl` project consists of many different sub-projects, such as a layer-based visualization framework, GPU toolkit, basemap react components, and more. The project is designed to be modular, so that it can be used in a variety of different environments and for different purposes.

> The scope of project includes documentation, testing, integration and the creation of other artifacts that aid the development, deployment, operation or adoption of the open source project.

Author: ```suggestion
```

This `vis.gl` Charter reflects a carefully constructed balanced role for the TSC and the CPC in the governance of the OpenJS Foundation. The charter amendment process is for the TSC to propose changes using simple majority of the full TSC, the proposed changes being subject to review and approval by the CPC. The CPC may additionally make amendments to the project charter at any time, though the CPC will not interfere with day-to-day discussions, votes, or meetings of the TSC.

### 2.1 Other Formal Project Relationships

Section Intentionally Left Blank

## Section 3: `vis.gl`'s TSC Governing Body

`vis.gl` is governed by its [Steering Committee Members](https://github.com/visgl/tsc/blob/master/README.md#steering-committee-members).

## Section 4: Roles & Responsibilities 

The roles and responsibilities of `vis.gl`'s TSC are described in [GOVERNANCE.md](./GOVERNANCE.md).

### Section 4.1 Project Operations & Management

Section Intentionally Left Blank

### Section 4.2: Decision-making, Voting, and/or Elections

> The vis.gl Technical Steering Committee (TSC) operates on a consensus-seeking basis. Most decisions are expected to be resolved through open discussion and agreement. However, when consensus cannot be reached, any TSC member may call for a vote.

Author: ```suggestion
The vis.gl TSC operates on a consensus-seeking basis. Most decisions are expected to be resolved through open discussion and agreement. However, when consensus cannot be reached, any TSC member may call for a vote.
```
